apps/omg_watcher/test/support/integration/test_helper.ex

```diff
@@ -51,6 +51,22 @@ defmodule OMG.Watcher.Integration.TestHelper do
     Process.sleep(100)
   end
 
+  @doc """
+  The above wait_for_block_fetch/2 function only waits for a block to appear in the
+  state and add some "random" sleep to give the database time to process and write the block.
+  This function will instead poll for block.get until found (or timeout).
+  """
+  def wait_for_block_inserted_in_db(block_nr, timeout) do
```

Comment:
here's my take on these helper functions. they're bloating the test suite and making the tests themselves *harder to read*. I prefer my tests self complete. Everything about a test is exactly in the test module.